### PR TITLE
Add methods to add refspecs and tests to view fetchspecs

### PR DESCRIPTION
**Pull request by kshyatt**
This should let you set a push refspec and have `LibGit2.push` do things.

### CODE REVIEW

**Comment by tkelman** [comment]
docstrings would be informative here (and also a mention in the push docstring maybe?)

**Comment by kshyatt** [comment]
Added docs with examples. The doc build succeeds locally for me.

File: base/libgit2/remote.jl [hunk 1]
```diff
@@ -104,6 +104,54 @@ function push_refspecs(rmt::GitRemote)
...
+
+julia> LibGit2.push_refspecs(remote)
+String["refs/heads/master"]
+```
+
+!!! note
+    You may need to [`close`](@ref) and reopen the `GitRemote`
+    in question after updating its push refspecs in order for
+    the change to take effect and for calls to [`push`](@ref)
+    to work.
```
Comment (ararslan) [inline base/libgit2/remote.jl]: I wonder if this is something we could handle behind the scenes?

Comment (kshyatt) [inline base/libgit2/remote.jl]: The plan is to get rid of `close` entirely so ideally soon we won't have to.

Comment (ararslan) [inline base/libgit2/remote.jl]: Okay

**Review by ararslan — APPROVED** [review]

**Comment by kshyatt** [comment]
Good to squash and merge?

**Pull request merged**